### PR TITLE
PR for 2.2_tensor demo 

### PR DESCRIPTION
**Pull request by GehangZhang**
**2.2Tensor介绍：**
[TODO]

- [x] 2.2.1 Tensor 是什么

- [x] 2.2.2 如何得到一个 Tensor

- [x] 2.2.3 Tensor 的常见属性

- [x] 2.2.4 常见的 Tensor 操作（算子）

### CODE REVIEW

**Comment by doombeaker** [comment]
这个分支应该往 `revision_demo_quickstart` 上合并，这样 reviewer 就不用重复看到 quickstart 那章已经有的内容

File: cn/docs/code/quick_start/log/default_physical_env_log/oneflow-15/oneflow.oneflow-15.zhanggehang.log.INFO.20210718-171030.12029 [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Log file created at: 2021/07/18 17:10:30
```
Comment (doombeaker) [inline cn/docs/code/quick_start/log/default_physical_env_log/oneflow-15/oneflow.oneflow-15.zhanggehang.log.INFO.20210718-171030.12029:1]: 不要又又又犯这种低级的错误好么

Comment (jackalcooper) [reply cn/docs/code/quick_start/log/default_physical_env_log/oneflow-15/oneflow.oneflow-15.zhanggehang.log.INFO.20210718-171030.12029]: 可以加入ignore？

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+
+
+## 什么是Tensor
+
+​	机器学习中，Tensor是我们离不开的数据形式。神经网络中数据表示，往往是通过称为张量（Tensor）的数据存储库完成的。张量（Tensor）可以视为一个容器，可以容纳N维数据。 我们可以将**Tensor**理解为多维数组，其可以具有任意多的维度，不同**Tensor**可以有不同的**数据类型** (dtype) 和**形状** (shape) 。
+
+​	张量类似于 NumPy 的 ndarray，不同之处在于张量可以在 GPU 或其他硬件加速器上运行。事实上，张量和 NumPy 数组通常可以共享相同的底层内存，从而无需复制数据，除此之外，张量也针对自动微分进行了优化。
+
+​	下面我们将使用Numpy创建一个张量：
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:11]: 上文提到 张量 **类似** ndarray，这里又说用 numpy 创建一个 **张量**。逻辑上是不对的：到底是 **类似**，还是**本来就是呢**。

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+Tensor可以通过 NumPy arrays进行构造。
+
+```python
+import oneflow as flow
+import numpy as np
+
+data = [[1, 2],[3, 4]]
+np_array = np.array(data)
+x_np = flow.from_numpy(np_array)
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:59]: 这个代码在 oneflow 能跑吗

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+
+
+
+
+## 如何得到一个Tensor
+
+
+
+### 使用Numpy构造（方法似乎还没更新？）
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:49]: PR 已经正式准备让人 review 了，这个括号里的东西是不应该存在的。
有疑问和提醒讨论的地方，可以放在 pr comment 里。

否则万一review时漏掉这里，这东西就直接见用户了

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+```
+
+
+
+### 使用Tensor中的方法构造
+
+我们可以使用oneflow中的Tensor方法构造张量。
+
+```python
+import oneflow.experimental as flow
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:69]: 这个不需要了
```
import oneflow as flow
```

File: cn/docs/basics_topics/quickstart.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
+# 快速上手
```
Comment (doombeaker) [inline cn/docs/basics_topics/quickstart.md:1]: 为什么这个PR里又 quick start 的内容。

File: cn/docs/code/quick_start/quickstart_demo.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import torch
```
Comment (doombeaker) [inline cn/docs/code/quick_start/quickstart_demo.py:1]: 同上，为什么要把多个PR的这些东西混到一起。

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+连接多个Tensor.
+
+```python
+t1 = torch.cat([tensor, tensor, tensor], dim=1)
+print(t1)
+```
+
+#### 兼有类
+
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:210]: 没写完

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+```
+
+连接多个Tensor.
+
+```python
+t1 = torch.cat([tensor, tensor, tensor], dim=1)
+print(t1)
+```
+
+#### 兼有类
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:209]: “兼有类”是东西？

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+  import numpy as np
+  flow.enable_eager_execution()
+  
+  x = flow.Tensor(np.array([-1, 2, -3, 4]).astype(np.float32))
+  x = flow.abs(x)
+  print('Tensor x:',X)
+  
+  ```
+
+  **是否要罗列所有数值计算类算子**
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:183]: 不要。另外这类东西请正式review后，请放到 comment 里讨论

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+  ```
+
+  
+
+- abs(y)
+
+  ```python
+  import oneflow.experimental as flow
+  import numpy as np
+  flow.enable_eager_execution()
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:175]: 不需要了

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+  x = flow.Tensor(np.random.randn(1,1))
+  y = flow.Tensor(np.random.randn(2,3))
+  out3 = flow.add(x, y).numpy()
+  print("out3 shape",out3.shape)
+  
+  ```
+
+  
+
+- abs(y)
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:170]: 这是什么鬼

File: cn/docs/basics_topics/quickstart.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+            print(f'Epoch [{epoch+1}/{num_epochs}], Step [{i+1}/{n_total_steps}], Loss: {loss.item():.4f}')
+
+```
+
+【输出效果】
+
+## 保存模型
+
+```python
+torch.save(model.state_dict(), "model.pth")
```
Comment (BBuf) [inline cn/docs/basics_topics/quickstart.md:107]: 这个pr多次出现了torch，而且翻译痕迹太严重啦？？

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+1. shape：描述了tensor的每个维度上元素的数量
+2. ndim： tensor的维度数量，例如vector的 ndim 为1，matrix的 ndim 为2.
+3. axis或者dimension：指tensor某个特定的维度
+
+
+
+
+**Tensor**的**dtype**
+
+我们可以通过Tensor.dtype来查看张量的数据类型，目前支持：'bool'，'float16'，'float32'，'float64'，'uint8'，'int8'，'int16'，'int32'，'int64'。
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:126]: oneflow 目前没有 `bool` 类型

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+print('First column: ', tensor[:, 0])
+print('Last column:', tensor[..., -1])
+tensor[:,1] = 0
+print(tensor)
+```
+
+连接多个Tensor.
+
+```python
+t1 = torch.cat([tensor, tensor, tensor], dim=1)
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:205]: 为什么是 torch

File: cn/docs/basics_topics/quickstart.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+print(model)
+```
+
+## 训练模型
+
+为了训练模型，我们需要 `loss` 函数和 `optimizer`，`loss` 函数用于评价神经网络预测的结果与 label 的差距；`optimizer` 调整网络的参数，使得网络预测的结果越来越接近 label（标准答案）。
+
+```python
+>>> loss_fn = nn.CrossEntropyLoss()
+>>> optimizer = torch.optim.SGD(model.parameters(), lr=1e-3)
```
Comment (doombeaker) [inline cn/docs/basics_topics/quickstart.md:73]: 为什么是 torch

File: cn/docs/basics_topics/2.2_Tensor.md [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+
+​	机器学习中，Tensor是我们离不开的数据形式。神经网络中数据表示，往往是通过称为张量（Tensor）的数据存储库完成的。张量（Tensor）可以视为一个容器，可以容纳N维数据。 我们可以将**Tensor**理解为多维数组，其可以具有任意多的维度，不同**Tensor**可以有不同的**数据类型** (dtype) 和**形状** (shape) 。
+
+​	张量类似于 NumPy 的 ndarray，不同之处在于张量可以在 GPU 或其他硬件加速器上运行。事实上，张量和 NumPy 数组通常可以共享相同的底层内存，从而无需复制数据，除此之外，张量也针对自动微分进行了优化。
+
+​	下面我们将使用Numpy创建一个张量：
+
+```python
+import numpy as np
+x = np.array([[[1, 4, 7],
```
Comment (doombeaker) [inline cn/docs/basics_topics/2.2_Tensor.md:15]: 张量这节代码都比较简单。我推荐都写成交互式的（doctest的那种），比如
```
>>> x = np.array(...)
>>> x
输出结果……
```
一个是让用户复制代码就可以运行（甚至之后可以直接放我们云平台上运行，像torch提供的云一样的）。

第二个，是后面我们要加上针对文档的测试，doctest框架也是通过搜索 `>>>` 来判断是否需要测试的。

File: cn/docs/code/quick_start/log/default_physical_env_log/oneflow-15/oneflow.oneflow-15.zhanggehang.log.INFO.20210718-171030.12029 (hunk 1 above)
Comment (doombeaker) [reply cn/docs/code/quick_start/log/default_physical_env_log/oneflow-15/oneflow.oneflow-15.zhanggehang.log.INFO.20210718-171030.12029]: 已经添加 .gitignore

Comment (doombeaker) [reply cn/docs/code/quick_start/log/default_physical_env_log/oneflow-15/oneflow.oneflow-15.zhanggehang.log.INFO.20210718-171030.12029]: 已经添加 .gitignore
